uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/ContactWorkPattern.php

```diff
@@ -222,4 +222,48 @@ public static function getAllForPeriod($contactID, DateTime $start, DateTime $en
     return $contactWorkPatterns;
   }
 
+  /**
+   * Gets the contacts that have work patterns overlapping the start and end date period.
+   *
+   * @param \DateTime $start
+   * @param \DateTime $end
+   * @param null|int $workPatternID
+   * @param bool $ignoreEffectiveDate
+   *   If true will return contacts who has Contact work patterns with effective_end date greater than
+   *   or equal to the start date parameter irrespective of what the effective/start date of the
+   *   Contact Work Pattern is.
+   *
+   * @return array
+   *   Array of contact ID's
+   */
+  public static function getContactsForPeriod(DateTime $start, DateTime $end, $workPatternID = null, $ignoreEffectiveDate = false) {
```

Comment:
You only use this method twice and on both you always set `$ignoreEffectiveDate` to true. I think it's better to completely remove it and make the method less complex.

Still regarding the method complexity, I can see you that on both times you've used the method, the WorkPattern ID is never null. Given that, I think you should rename the method to something like `getContactsUsingWorkPatternInPeriod` and make the `$workPatternID` param mandatory.